### PR TITLE
use @arbitrum/sdk  / update eth deposit check

### PR DESCRIPTION
**Pull request by DZGoldman**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -333,15 +348,13 @@ function App() {
 
     setL1TxnHashState(L1ReceiptState.MESSAGES_FOUND);
 
-    const looksLikeEthDeposit = await l1TxnReceipt.looksLikeEthDeposit(
-      l1Provider
-    );
+    const isDefinitelyEthDeposit = await isArbInboxEthDeposit(l1Provider,txHash)
```
Comment (fredlacs) [inline src/App.tsx:351]: are we not worried of false negatives here? ie smart contract eth deposits

File: src/App.tsx [hunk 2]
```diff
@@ -141,6 +140,22 @@ const getL1TxnReceipt = async (txnHash: string) => {
   }
 };
 
+const isArbInboxEthDeposit = async(
```
Comment (fredlacs) [inline src/App.tsx:143]: We should probably add a note saying this doesn't detect all eth deposits

Comment (fredlacs) [reply src/App.tsx:143]: other people looking into this repo for inspiration might shoot themselves on the foot

**Review by fredlacs — CHANGES_REQUESTED** [review]
The changes here overlap a bit with my other PR in https://github.com/OffchainLabs/retryable-tx-panel/pull/8

The main refactor done there which I think is beneficial that isn't present here is handling chain IDs as numbers instead of string

File: src/App.tsx [hunk 3]
```diff
@@ -38,6 +39,11 @@ interface L1ToL2MessageReaderWithNetwork extends L1ToL2MessageReader {
   l2Network: L2Network;
 }
 
+const isEthDeposit = async (l1ToL2Message:L1ToL2MessageReader ): Promise<boolean>=>{
+  const txData = (await l1ToL2Message.l2Provider.getTransaction(l1ToL2Message.retryableCreationId)).data
+  // check that calldataSize param is zero (8th 32-byte param, offset by 4 bytes for message ID):
```
Comment (DZGoldman) [inline src/App.tsx:44]: I think this works / am going off of https://github.com/OffchainLabs/arb-os/blob/962ab51cffff5d57daee0f4b031d69a3224d51d4/arb_os/retrybuffer.mini#L305

could use a sanity check tho

File: src/App.tsx (hunk 2 above)
Comment (DZGoldman) [reply src/App.tsx]: updated detection method

**Review by yahgwai — APPROVED** [review]

**Review by yahgwai — APPROVED** [review]

**Pull request merged**